Bookish.DataAccess/BookishService.cs

```diff
@@ -85,7 +87,39 @@ public IEnumerable<BookLoan> GetLoans(string userId)
                 ORDER BY
 	                Loans.DueDate";
 
-            return dbConnection.Query<BookLoan>(query, new { UserID = userId });
+            return dbConnection.Query<BookLoan>(query, new {UserID = userId});
+        }
+
+        public void AddBook(string title, string author, string isbn, int totalCopies)
+        {
+            var booksQuery =
+                @"INSERT INTO Books (ISBN, Title, Author)
+                VALUES (@ISBN, @Title, @Author)";
+
+            var parameters = new { Title = title, Author = author, ISBN = isbn };
+
+            dbConnection.Execute(booksQuery, parameters);
+
+            var lastId = LastCopiesId();
+
+            var valuesList = new List<string>();
+            for (var i = 1; i <= totalCopies; i++)
+            {
+                valuesList.Add($"({lastId + i}, @ISBN)");
+            }
+
+            var copiesQuery =
+                @"INSERT INTO BookCopies (CopyId, ISBN) VALUES " +
+                string.Join(", ", valuesList);
```

Comment:
You should be able to do this all as one query 🙂

Reply:
Do you mean doing insertions into both tables using one query?

Reply:
Yes. You'll need two insert statements but you should be able to do it all in sql. Most importantly if the first insert fails then the second shouldn't happen. So make sure to test that case 🙂